doc/02_development/09_actions.md

```diff
@@ -0,0 +1,127 @@
+# Github actions
+
+**Summary:** This page explains the GitHub actions we use to unsure Code quality.
+
+---
+
+## Author
+
+Tim Dreier
+
+## Date
+
+25.11.2022
+
+## Table of contents
+<!-- TOC -->
+* [Github actions](#github-actions)
+  * [Author](#author)
+  * [Date](#date)
+  * [Table of contents](#table-of-contents)
+  * [General](#general)
+  * [Pull requests](#pull-requests)
+  * [Steps](#steps)
+    * [1. Check out the repo](#1-check-out-the-repo)
+      * [Checkout definition in yaml file](#checkout-definition-in-yaml-file)
+    * [2. Run the markdown linter](#2-run-the-markdown-linter)
+      * [Markdown lint definition in yaml file](#markdown-lint-definition-in-yaml-file)
+    * [3. Run the python linter](#3-run-the-python-linter)
+      * [Python lint definition in yaml file](#python-lint-definition-in-yaml-file)
+  * [Sources](#sources)
+<!-- TOC -->
+
+## General
+
+We use a github action to verify code quality.
+These actions are defined in `.github/workflows/linter.yml`.
+
+The actions are executed only on pull requests in order not to exceed the [minutes per month included in the Github](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) free plan.
+This is done by limiting the execution of the action by the following line:
+
+```yaml
+on: pull_request
+```
+
+The actions uses the same linters described in the section [Linting](./02_linting.md).
+
+Event though the linters are already executed during commit,
+the execution on pull request ensures that nobody skips the linter during commit.
+
+## Pull requests
+
+The action must pass successful to merge a pull request.
+To enforce this behaviour, we set the action as requirement as described in the following:
+
+> You can add rules to your branches and how your merge request works.
+>
+> Your repo -> Settings -> branches -> Branch protection rules -> Add rule
+>
+> There, you will find something called Require status checks to pass before merging. Under this, you should see Status checks found in the last week for this repository. If you find the status you want to be passed before merging, you can enable the same.
+>
+> Docs: <https://help.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests>
+>
+> [(Source)](https://stackoverflow.com/questions/60776412/github-actions-is-there-a-way-to-make-it-mandatory-for-pull-request-to-merge)
+
+More information about creating and merging pull requests can be found [here](./08_project_management.md).
+
+## Steps
```

Comment:
Do you think it is necessary to include the steps of the github actions file? I think I would rather resort to having it link to the actual .yaml file - (maybe add some comments there)?
--> To have a single source of truth

Reply:
Resolved :)